src/entry.c

```diff
@@ -326,6 +326,11 @@ gboolean CacheEntryPasswordSet(CacheEntry *self, const gchar *password,
   return TRUE;
 }
 
+void CacheEntryHashSet(CacheEntry *self, CacheEntryAlgorithm algorithm,
+                  GBytes *hash, CacheEntryArgs *args) {
```

Comment:
Sorry, yes you are increasing the ref count correctly.

Also, this should have a comment that looks like the other public functions. The actual format is described in the section of the gtk-doc manual: https://developer.gnome.org/gtk-doc-manual/unstable/documenting.html.en